app/javascript/react/components/LocationSearch/LocationSearch.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect } from "react";
+import usePlacesAutocomplete, {
+  getGeocode,
+  getLatLng,
+} from "use-places-autocomplete";
+import { useCombobox } from "downshift";
+
+import { LatLngLiteral } from "../../types/googleMaps";
+import * as S from "./LocationSearch.style";
+import locationSearchIcon from "../../assets/icons/locationSearchIcon.svg";
+import { SearchContainer } from "../Navbar/Navbar.style";
+
+interface LocationSearchProps {
+  setLocation: (position: LatLngLiteral) => void;
+  isMapPage?: boolean;
+}
+
+type AutocompletePrediction = google.maps.places.AutocompletePrediction;
+
+const LocationSearch = ({ setLocation, isMapPage }: LocationSearchProps) => {
```

Comment:
I see that we're passing `isMapPage` depending on the context Mobile/Desktop header. Is it connected with the map page or rather a mobile view? 🤔